### ISSUE TITLE
Disable upload of additional files once a file has been sent and before it has returned

### ISSUE THREAD

**Issue by NAnguiano** [dondi/GRNsight#475]
As I noted in #474, once a file is uploaded, you're capable of continuing to try and upload files even if the one you uploaded hasn't returned yet. They seem to just build in some sort of stack until whatever is blocking them from loading finishes loading and then they all load at once. 

So, I propose disabling the upload of additional files once a file has been uploaded, and re-enabling it once the file has returned. Some sort of "loading" indicator (for example, on the taskbar) could also be useful.

**Comment by kdahlquist** [comment]
That sounds reasonable.

**Comment by dondi** [comment]
@mihirsamdarshi and @dondi discussed the implementation approach for this and @mihirsamdarshi will execute it this week.

**Comment by mihirsamdarshi** [comment]
Looking at this further, I actually believe that this is already actually handled. There is a `event.preventDefault()` function in the `uploadEpilogue` function that is run in the upload handler. This is contained on lines 25-51 in the [`setup-load-and-import-handlers.js`](https://github.com/dondi/GRNsight/blob/40f2e93e11983944e2b6e94854e15c77efde6a69/web-client/public/js/setup-load-and-import-handlers.js#L25) file.

**Comment by dondi** [comment]
I don’t believe that `event.preventDefault()` is doing that. This isn’t what `event.preventDefault()` does. What it does is keep the browser from doing what it _normally_ does when that event happens.

Tracing the code, `uploadEpilogue` is called by `uploadHandler`, which in turn is set as the [handler for the `change` event](https://github.com/dondi/GRNsight/blob/40f2e93e11983944e2b6e94854e15c77efde6a69/web-client/public/js/setup-load-and-import-handlers.js#L135-L137) on the upload controls.

Those upload controls are `a` elements, which means that their default behavior is for the browser to "navigate" to those locations. _That_ is the default behavior which is being prevented. We aren’t using those elements as web links but just as menu items. It has nothing to do with preventing an upload when it's already happening.

**Comment by mihirsamdarshi** [comment]
Sorry @dondi, I commented before tracing it. However, I figured out that I could disable the button before calling the upload handler, and then re-enabling it in the AJAX success function. However, I ran into a weird little issue. The current upload code on line 142 of [`setup-load-and-import-handlers.js`](https://github.com/dondi/GRNsight/blob/40f2e93e11983944e2b6e94854e15c77efde6a69/web-client/public/js/setup-load-and-import-handlers.js#L135) is the following: 
```Javascript
$("#upload").on("change", uploadHandler("upload", loadGrn));
```
however, in order to add the disable button function I created, it needs to either look like:
```Javascript
$("#upload").on("change", function () {
    uploadHandler("upload", loadGrn);
});
```
or
```Javascript
$("#upload").change(() => {
    uploadHandler("upload", loadGrn);
});
```
However, when I change it to either of these, the upload function stops working. I haven't run into syntactical errors very often, so I am not sure what to look up online.

**Comment by dondi** [comment]
Hint: What does `uploadHandler` return?

**Comment by mihirsamdarshi** [comment]
@dondi thanks for pointing that out, this has been implemented in #796

**Comment by kdahlquist** [comment]
I don't know how to test this.  Maybe on a slow internet connection?

**Comment by dondi** [comment]
Replication would have to be done through a temporary artificial delay:
* Either temporarily throttle the server code
* Or (if it is sufficient) use Chrome's ability to artificially simulate a super slow network connection.

@dondi and @afiller1 to verify.

**Comment by kdahlquist** [comment]
This is still an outstanding review requested for @dondi or @afiller1.  It would be great if this could be independently verified so we can close this issue.

**Comment by kdahlquist** [comment]
@dondi, this still needs to be reviewed and closed.

**Comment by kdahlquist** [comment]
@dondi, can you review this one at your next work session?

**Comment by dondi** [comment]
Yes, I'll line this up for this week’s batch of issues.

**Comment by dondi** [comment]
@dondi will give this one last try in terms of replicating/verifying on a firsthand basis (particularly, trying via the Chrome dev tools) and will report back.

**Comment by dondi** [comment]
I was able to verify this directly by activating Google Chrome’s simulated network throttling feature, which allows connections to behave much slower than they normally would. The lowest setting of Slow 3G was used:

<img width="194" alt="image" src="https://user-images.githubusercontent.com/382242/95523796-7056c580-0984-11eb-8e2d-cdbdc5b58372.png">

With this setting, I was able to attempt to upload another file while the first upload was still in progress. Those attempts all failed, as they should. A subsequent upload could not be performed until the current upload-in-progress was completely finished.

I’ll go ahead and close this, given this finding.
